### ISSUE TITLE
manual/javaGuide/main/http/JavaSessionFlash.md

### ISSUE THREAD

**Issue by garbagetown** [playframework-ja/translation-project#956]
- https://github.com/playframework-ja/translation-project/tree/2.3.5/manual/javaGuide/main/http/JavaSessionFlash.md
- diffs between [2.2.0/manual/javaGuide/main/http/JavaSessionFlash.md](https://github.com/playframework-ja/translation-project/tree/2.2.0/manual/javaGuide/main/http/JavaSessionFlash.md) and [2.3.5/manual/javaGuide/main/http/JavaSessionFlash.md](https://github.com/playframework-ja/translation-project/tree/2.3.5/manual/javaGuide/main/http/JavaSessionFlash.md).

```
--- /Users/garbagetown/Desktop/2.2.0/manual/javaGuide/main/http/JavaSessionFlash.md 2013-07-29 23:52:24.000000000 +0900
+++ //Users/garbagetown/Desktop/2.3.5/manual/javaGuide/main/http/JavaSessionFlash.md    2014-07-28 12:26:38.000000000 +0900
@@ -1,3 +1,4 @@
+<!--- Copyright (C) 2009-2013 Typesafe Inc. <http://www.typesafe.com> -->
 # Session and Flash scopes

 ## How it is different in Play
@@ -10,51 +11,27 @@

 > There is no technical timeout for the session, which expires when the user closes the web browser. If you need a functional timeout for a specific application, just store a timestamp into the user Session and use it however your application needs (e.g. for a maximum session duration, maxmimum inactivity duration, etc.).

-## Reading a Session value
-
-You can retrieve the incoming Session from the HTTP request:
-
-'''java
-public static Result index() {
-  String user = session("connected");
-  if(user != null) {
-    return ok("Hello " + user);
-  } else {
-    return unauthorized("Oops, you are not connected");
-  }
-}
-'''
-
 ## Storing data into the Session

 As the Session is just a Cookie, it is also just an HTTP header, but Play provides a helper method to store a session value:

-'''java
-public static Result index() {
-  session("connected", "user@gmail.com");
-  return ok("Welcome!");
-}
-'''
+@[store-session](code/javaguide/http/JavaSessionFlash.java)

 The same way, you can remove any value from the incoming session:

-'''java
-public static Result index() {
-  session().remove("connected");
-  return ok("Bye");
-}
-'''
+@[remove-from-session](code/javaguide/http/JavaSessionFlash.java)
+
+## Reading a Session value
+
+You can retrieve the incoming Session from the HTTP request:
+
+@[read-session](code/javaguide/http/JavaSessionFlash.java)

 ## Discarding the whole session

 If you want to discard the whole session, there is special operation:

-'''java
-public static Result index() {
-  session().clear();
-  return ok("Bye");
-}
-'''
+@[discard-whole-session](code/javaguide/http/JavaSessionFlash.java)

 ## Flash scope

@@ -65,23 +42,12 @@

 > **Important:** The flash scope should only be used to transport success/error messages on simple non-Ajax applications. As the data are just kept for the next request and because there are no guarantees to ensure the request order in a complex Web application, the Flash scope is subject to race conditions.

-Here are a few examples using the Flash scope:
+So for example, after saving an item, you might want to redirect the user back to the index page, and you might want to display an error on the index page saying that the save was successful.  In the save action, you would add the success message to the flash scope:
+
+@[store-flash](code/javaguide/http/JavaSessionFlash.java)

-'''java
-public static Result index() {
-  String message = flash("success");
-  if(message == null) {
-    message = "Welcome!";
-  }
-  return ok(message);
-}
-
-public static Result save() {
-  flash("success", "The item has been created");
-  return redirect("/home");
-}
-'''
+Then in the index action, you could check if the success message existed in the flash scope, and if so, render it:

-&nbsp;
+@[read-flash](code/javaguide/http/JavaSessionFlash.java)

 > **Next:** [[Body parsers | JavaBodyParsers]]
\ No newline at end of file

```

**Comment by garbagetown** [comment]
:hand:

**Comment by garbagetown** [comment]
merged at #1125
